spec/lib/http_spec.rb

```diff
@@ -456,6 +456,15 @@ def setsockopt(*args)
 
         expect(response.to_s).to eq("")
       end
+
+      it "returns decoded body with the correct charset" do
+        encoding = Encoding::Shift_JIS
+        client   = HTTP.use(:auto_inflate).headers("Accept-Encoding" => "gzip").encoding(encoding)
+        body     = "もしもし!".encode(encoding, Encoding::UTF_8)
+        response = client.post("#{dummy.endpoint}/encoded-body", :body => body)
```

Comment:
I'm struggling to understand this test.

``` ruby
client   = HTTP.use(:auto_inflate).headers("Accept-Encoding" => "gzip").encoding(encoding)
```
^^^ `encoding(...)` enforces encoding of the response.